test: speed up remote tests

ldorau opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/pmem/nvml/1060)
<!-- Reviewable:end -->

plebioda commented:
I get the following errors when running tests:
./RUNTESTS rpmem_fip
rpmem_fip/TEST0: SETUP (check/none/debug/verbs/GPSPM)
bash: line 0: cd: /dev/shm/node0/test_rpmem_fip: No such file or directory

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1060#-:-KQQK07UCMHGZnhKELDZ:b-dr9diz)*
<!-- Sent from Reviewable.io -->

plebioda commented:
Reviewed 1 of 57 files at r1.
Review status: 1 of 61 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1060)*
<!-- Sent from Reviewable.io -->

krzycz commented:
Reviewed 54 of 57 files at r1, 18 of 18 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[src/test/sync-remotes/copy-to-remote-nodes.sh, line 34 \[r2\]](https://reviewable.io:443/reviews/pmem/nvml/1060#-KQQTMgamXnqe_wqffol:-KQQT9uk6IP9s1zdQ5dx:bj440mw) ([raw file](https://github.com/pmem/nvml/blob/36c337a77d3df69266a923619eaf44eb9e2b7ad1/src/test/sync-remotes/copy-to-remote-nodes.sh#L34)):*
> ```Shell
> #
> 
> UNITTEST_NAME=0
> ```

So, is this a test or some helper script that is used by other tests?
A comment?

---

*[src/test/sync-remotes/Makefile, line 34 \[r2\]](https://reviewable.io:443/reviews/pmem/nvml/1060#-KQQS7mpQ1_Okc1LUWvB:-KQQS7mpQ1_Okc1LUWvC:b-mhhohq) ([raw file](https://github.com/pmem/nvml/blob/36c337a77d3df69266a923619eaf44eb9e2b7ad1/src/test/sync-remotes/Makefile#L34)):*
> ```Makefile
> 
> #
> # src/test/sync-remote/Makefile --
> ```

missing description?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1060)*
<!-- Sent from Reviewable.io -->

ldorau commented:
You should run 'make test' first (which calls 'make sync-remotes') or  'make sync-remotes' in src/test directly.

---

Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1060#-:-KQQUjAR_EjlrunOu6f_:b-a9trw6)*
<!-- Sent from Reviewable.io -->

ldorau commented:
Wait, my mistake - 'make check' calls 'make sync-remotes' ....

---

Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1060#-:-KQQVcnUAj7EWaOTPNcI:b-bovqu3)*
<!-- Sent from Reviewable.io -->

ldorau commented:
Now "make test" calls "make sync-remotes"

---

Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[src/test/sync-remotes/copy-to-remote-nodes.sh, line 34 \[r2\]](https://reviewable.io:443/reviews/pmem/nvml/1060#-KQQTMgamXnqe_wqffol:-KQQikC6w166wly6lCUQ:b-rtpbah) ([raw file](https://github.com/pmem/nvml/blob/36c337a77d3df69266a923619eaf44eb9e2b7ad1/src/test/sync-remotes/copy-to-remote-nodes.sh#L34)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary>
> So, is this a test or some helper script that is used by other tests?
> A comment?
</details>
A comment added.

---

*[src/test/sync-remotes/Makefile, line 34 \[r2\]](https://reviewable.io:443/reviews/pmem/nvml/1060#-KQQS7mpQ1_Okc1LUWvB:-KQQjFy7feYvAp20l77Z:b-8b11j9) ([raw file](https://github.com/pmem/nvml/blob/36c337a77d3df69266a923619eaf44eb9e2b7ad1/src/test/sync-remotes/Makefile#L34)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary>
> missing description?
</details>
Added

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1060#-:-KQQjJL_kspacwIkVokx:bew0ev1)*
<!-- Sent from Reviewable.io -->

krzycz commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 6 of 6 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1060#-:-KQRRPvXKTvQb3uX06Qb:bnfp4nl)*
<!-- Sent from Reviewable.io -->

plebioda commented:
*[Makefile, line 39 \[r3\]](https://reviewable.io:443/reviews/pmem/nvml/1060#-KQVqoTsopHu6g5Jy_fz:-KQVqoTsopHu6g5Jy_g-:b2w7ung) ([raw file](https://github.com/pmem/nvml/blob/600e3e9b80d8606c7ff457d9f9f14438207a1c1d/Makefile#L39)):*
> ```Makefile
> # Use "make doc" to build documentation.
> #
> # Use "make test" to build unit tests. Add "SKIP_SYNC_REMOTES=y" to skip syncing
> ```

I would add something like FORCE_SYNC_REMOTES=y. Sometimes you might be out of sync if you modify something on the remote dirs and then it would be much easier to run make test FORCE_SYNC_REMOTES=y than make clobber and make test.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1060)*
<!-- Sent from Reviewable.io -->

plebioda commented:
*[src/test/Makefile.inc, line 272 \[r3\]](https://reviewable.io:443/reviews/pmem/nvml/1060#-KQVvgyioswCneNcrZ9R:-KQVvgyjC4Ae0rxz187o:bbob36g) ([raw file](https://github.com/pmem/nvml/blob/600e3e9b80d8606c7ff457d9f9f14438207a1c1d/src/test/Makefile.inc#L272)):*
> ```Makefile
> $(TESTCONFIG):
> 
> sync-test: all $(SYNC_FILE)
> ```

I wonder if it would be better to add sync-test to default rule, so when you run make inside test dir it will be synchronized with remote node... But I'm not sure..

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1060)*
<!-- Sent from Reviewable.io -->

plebioda commented:
*[src/test/Makefile.inc, line 275 \[r3\]](https://reviewable.io:443/reviews/pmem/nvml/1060#-KQVzzhp8QRYLbfGC0Tn:-KQVzzhp8QRYLbfGC0To:b-d4byqu) ([raw file](https://github.com/pmem/nvml/blob/600e3e9b80d8606c7ff457d9f9f14438207a1c1d/src/test/Makefile.inc#L275)):*
> ```Makefile
> 
> $(SYNC_FILE): $(TARGET) $(TESTCONFIG)
> ifeq ($(SCP_TO_REMOTE_NODES), yes)
> ```

y ? it's more common

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1060)*
<!-- Sent from Reviewable.io -->

plebioda commented:
*[src/test/sync-remotes/copy-to-remote-nodes.sh, line 51 \[r3\]](https://reviewable.io:443/reviews/pmem/nvml/1060#-KQW0OO6K8SVMUq51R7t:-KQW0OO6K8SVMUq51R7u:bq8grin) ([raw file](https://github.com/pmem/nvml/blob/600e3e9b80d8606c7ff457d9f9f14438207a1c1d/src/test/sync-remotes/copy-to-remote-nodes.sh#L51)):*
> ```Shell
> 
> case "$COPY_TYPE" in
> 	commons)
> ```

common ?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1060)*
<!-- Sent from Reviewable.io -->

plebioda commented:
*[src/test/tools/Makefile, line 56 \[r3\]](https://reviewable.io:443/reviews/pmem/nvml/1060#-KQW0beWjeJGHzZ--SR5:-KQW0beWjeJGHzZ--SR6:b5dne8c) ([raw file](https://github.com/pmem/nvml/blob/600e3e9b80d8606c7ff457d9f9f14438207a1c1d/src/test/tools/Makefile#L56)):*
> ```Makefile
> all test cstyle clean clobber format: $(DIRS)
> 
> sync-remotes: ctrld fip
> ```

I would add some variable like REMOTE_TOOLS ?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1060)*
<!-- Sent from Reviewable.io -->

plebioda commented:
*[src/test/unittest/unittest.sh, line 2050 \[r3\]](https://reviewable.io:443/reviews/pmem/nvml/1060#-KQW5OfZOg_c4pt5dsqc:-KQW5OfZOg_c4pt5dsqd:b-16lqsl) ([raw file](https://github.com/pmem/nvml/blob/600e3e9b80d8606c7ff457d9f9f14438207a1c1d/src/test/unittest/unittest.sh#L2050)):*
> ```Shell
> 
> 	local NODES_MAX=$((${#NODE[@]} - 1))
> 	local NODES_SEQ=$(seq -s' ' 0 $NODES_MAX)
> ```

isn't this variable in global scope ?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1060)*
<!-- Sent from Reviewable.io -->

plebioda commented:
*[src/test/unittest/unittest.sh, line 2076 \[r3\]](https://reviewable.io:443/reviews/pmem/nvml/1060#-KQW66lgYjV8l56aerOE:-KQW66lgYjV8l56aerOF:b-dqp4pd) ([raw file](https://github.com/pmem/nvml/blob/600e3e9b80d8606c7ff457d9f9f14438207a1c1d/src/test/unittest/unittest.sh#L2076)):*
> ```Shell
> 			&& exit 1
> 
> 		# check if the node is reachable
> ```

separate function ?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1060)*
<!-- Sent from Reviewable.io -->

plebioda commented:
*[src/test/unittest/unittest.sh, line 2057 \[r3\]](https://reviewable.io:443/reviews/pmem/nvml/1060#-KQW6EGI1L06csftww_u:-KQW6EGI1L06csftww_v:b-dqp4pd) ([raw file](https://github.com/pmem/nvml/blob/600e3e9b80d8606c7ff457d9f9f14438207a1c1d/src/test/unittest/unittest.sh#L2057)):*
> ```Shell
> 		&& exit 1
> 
> 	# add all libraries to the 'to-copy' list
> ```

separate function ?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1060)*
<!-- Sent from Reviewable.io -->

plebioda commented:
Reviewed 40 of 57 files at r1, 17 of 18 files at r2, 6 of 6 files at r3.
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1060)*
<!-- Sent from Reviewable.io -->

ldorau commented:
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[Makefile, line 39 \[r3\]](https://reviewable.io:443/reviews/pmem/nvml/1060#-KQVqoTsopHu6g5Jy_fz:-KQ_LtcHe8xakEYEzT5U:b-8b11j9) ([raw file](https://github.com/pmem/nvml/blob/600e3e9b80d8606c7ff457d9f9f14438207a1c1d/Makefile#L39)):*
<details><summary><i>Previously, plebioda wrote…</i></summary>
> I would add something like FORCE_SYNC_REMOTES=y. Sometimes you might be out of sync if you modify something on the remote dirs and then it would be much easier to run make test FORCE_SYNC_REMOTES=y than make clobber and make test.
</details>
Added

---

*[src/test/Makefile.inc, line 272 \[r3\]](https://reviewable.io:443/reviews/pmem/nvml/1060#-KQVvgyioswCneNcrZ9R:-KQW48ZrNITjCu1PCfKY:bivi5gp) ([raw file](https://github.com/pmem/nvml/blob/600e3e9b80d8606c7ff457d9f9f14438207a1c1d/src/test/Makefile.inc#L272)):*
<details><summary><i>Previously, plebioda wrote…</i></summary>
> I wonder if it would be better to add sync-test to default rule, so when you run make inside test dir it will be synchronized with remote node... But I'm not sure..
</details>
postponed

---

*[src/test/unittest/unittest.sh, line 2050 \[r3\]](https://reviewable.io:443/reviews/pmem/nvml/1060#-KQW5OfZOg_c4pt5dsqc:-KQW6duEMBydDYI9TdL1:bhb7bs1) ([raw file](https://github.com/pmem/nvml/blob/600e3e9b80d8606c7ff457d9f9f14438207a1c1d/src/test/unittest/unittest.sh#L2050)):*
<details><summary><i>Previously, plebioda wrote…</i></summary>
> isn't this variable in global scope ?
</details>
It is another variable - it includes all nodes listed in the config file. I have changed its name.

---

*[src/test/unittest/unittest.sh, line 2057 \[r3\]](https://reviewable.io:443/reviews/pmem/nvml/1060#-KQW6EGI1L06csftww_u:-KQ_jD86UNgs-gPes6JU:b-8b11j9) ([raw file](https://github.com/pmem/nvml/blob/600e3e9b80d8606c7ff457d9f9f14438207a1c1d/src/test/unittest/unittest.sh#L2057)):*
<details><summary><i>Previously, plebioda wrote…</i></summary>
> separate function ?
</details>
Added

---

*[src/test/unittest/unittest.sh, line 2076 \[r3\]](https://reviewable.io:443/reviews/pmem/nvml/1060#-KQW66lgYjV8l56aerOE:-KQ_jEbYnCo0vLL0qMUT:b-8b11j9) ([raw file](https://github.com/pmem/nvml/blob/600e3e9b80d8606c7ff457d9f9f14438207a1c1d/src/test/unittest/unittest.sh#L2076)):*
<details><summary><i>Previously, plebioda wrote…</i></summary>
> separate function ?
</details>
Added

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1060)*
<!-- Sent from Reviewable.io -->

krzycz commented:
Reviewed 24 of 24 files at r4.
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1060)*
<!-- Sent from Reviewable.io -->

plebioda commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1060#-:-KQeyYUHFwTG5vnDgGs0:bnfp4nl)*
<!-- Sent from Reviewable.io -->
